src/app/common/services/unit-service.coffee

```diff
@@ -318,6 +318,45 @@ angular.module("doubtfire.common.services.units", [])
       else
         'No Tutorial'
 
+    student.calcTopTasks = () ->
+      #
+      # sort tasks by start date
+      #
+      sortedTasks = _.sortBy(_.sortBy(_.filter(student.tasks, (task) -> _.includes taskService.validTopTask, task.status), 'definition.seq'), 'definition.start_date')
```

Comment:
We want to avoid using taskService for things like this if we can. Instead of calling this call `task.isValidTopTask()` then adjust the code in task so that it has this new method. That method can call the task service for the moment. 

There are a number of comments all related to this.

Reply:
Ok. I will do this change.